lib/smart_proxy_container_gateway/container_gateway_main.rb

```diff
@@ -82,6 +83,20 @@ def authorized_for_repo?(repo_name)
         !unauthenticated_repo.nil?
       end
 
+      def valid_token?(token)
+        tokens = initialize_db[:authentication_tokens]
+        tokens.where(token_checksum: Digest::SHA256.hexdigest(token)).where do
+          expire_at > Sequel::CURRENT_TIMESTAMP
+        end.count.positive?
+      end
+
+      def insert_token(username, token, expire_at_string, clear_tokens: true)
+        tokens = initialize_db[:authentication_tokens]
+
+        tokens.insert(username: username, token_checksum: Digest::SHA256.hexdigest(token), expire_at: expire_at_string.to_s)
+        tokens.where { expire_at < Sequel::CURRENT_TIMESTAMP }.delete if clear_tokens
```

Comment:
Do you think it's worth also deleting unexpired tokens that have the same `token_checksum`?  I can't imagine that duplicate tokens from logging in and out would become a problem, but they are unneeded (I'm pretty sure?) duplicates regardless.

Reply:
good question, it seemed kinda weird that katello was handing out tokens again (and extending their expiration time).  Seems simple enough to delete existing tokens